test/request_reply_message_publisher_test.go

```diff
@@ -130,4 +134,421 @@ var _ = Describe("RequestReplyPublisher", func() {
 
 	})
 
+	Context("with a connected messaging service", func() {
+		var messageBuilder solace.OutboundMessageBuilder
+		BeforeEach(func() {
+			helpers.ConnectMessagingService(messagingService)
+			messageBuilder = messagingService.MessageBuilder()
+		})
+
+		AfterEach(func() {
+			if messagingService.IsConnected() {
+				helpers.DisconnectMessagingService(messagingService)
+			}
+		})
+
+		It("can print the publisher to a string and see the pointer", func() {
+			publisher, err := messagingService.RequestReply().CreateRequestReplyMessagePublisherBuilder().Build()
+			Expect(err).ToNot(HaveOccurred())
+			str := fmt.Sprint(publisher)
+			Expect(str).ToNot(BeEmpty())
+			Expect(str).To(ContainSubstring(fmt.Sprintf("%p", publisher)))
+		})
+
+		Describe("request-reply publisher termination tests", func() {
+			publishTopic := resource.TopicOf("hello/world")
+			topicSubscription := resource.TopicSubscriptionOf("hello/world")
+			largeByteArray := make([]byte, 16384)
+			timeOut := 5 * time.Second
+
+			// A handler for the request-reply publisher
+			publisherReplyHandler := func(message message.InboundMessage, userContext interface{}, err error) {
+				if err == nil { // Good, a reply was received
+					Expect(message).ToNot(BeNil())
+					payload, _ := message.GetPayloadAsString()
+					fmt.Printf("The reply inbound payload: %s\n", payload)
```

Comment:
Why is there is a print statement the individual test should not be printing to standard out directly except in debugging scenarios. Should this be a an expect statement. Note all other uses of fmt.print like function are used for str concatenation not printing to standard out. (Except in test setup and tear down of test infrastructure. ie running ouside the tests it self)

Reply:
Yeah, I used it for local testing and forgot to remove it before committing my changes.